arviz/data/inference_data.py

```diff
@@ -464,10 +490,10 @@ def to_netcdf(
                         for var_name, values in data.variables.items()
                         if _compressible_dtype(values.dtype)
                     }
-                data.to_netcdf(filename, mode=mode, group=group, **kwargs)
+                data.to_netcdf(filename, mode=mode, group=f"{base_group}/{group}", **kwargs)
                 data.close()
                 mode = "a"
-        elif not self._attrs:  # creates a netcdf file for an empty InferenceData object.
+        elif not self._attrs and base_group == '/':  # creates a netcdf file for an empty InferenceData object.
```

Comment:
I am not sure about the logic here. If the inferencedata is empty, a file is still created, but it is an empty file. Shouldn't that happen if base_group =! "/"? I guess ideally a file with that group present as an empty dataset, otherwise probably an empty file to preserve the "to_netcdf always creates a file, even if saving an empty inferencedata object" behaviour

Reply:
Sounds fair enough. I've restored the previous behavior.